processor/k8sattributesprocessor/processor.go

```diff
@@ -151,46 +151,64 @@ func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pco
 // addContainerAttributes looks if pod has any container identifiers and adds additional container attributes
 func (kp *kubernetesprocessor) addContainerAttributes(attrs pcommon.Map, pod *kube.Pod) {
 	containerName := stringAttributeFromMap(attrs, conventions.AttributeK8SContainerName)
-	if containerName == "" {
-		return
-	}
-	containerSpec, ok := pod.Containers[containerName]
-	if !ok {
-		return
-	}
-
-	if containerSpec.ImageName != "" {
-		if _, found := attrs.Get(conventions.AttributeContainerImageName); !found {
-			attrs.PutStr(conventions.AttributeContainerImageName, containerSpec.ImageName)
+	containerID := stringAttributeFromMap(attrs, conventions.AttributeContainerID)
+	var (
+		containerSpec *kube.Container
+		ok            bool
+	)
+	switch {
+	case containerName != "":
+		containerSpec, ok = pod.Containers.ByName[containerName]
+		if !ok {
+			return
 		}
-	}
-	if containerSpec.ImageTag != "" {
-		if _, found := attrs.Get(conventions.AttributeContainerImageTag); !found {
-			attrs.PutStr(conventions.AttributeContainerImageTag, containerSpec.ImageTag)
+	case containerID != "":
+		containerSpec, ok = pod.Containers.ByID[containerID]
+		if !ok {
+			return
 		}
+	default:
+		return
 	}
-
```

Comment:
Are all the changes below needed for this change or just refactoring?

Reply:
It's basically a refactoring. I tried to avoid duplicating that code even more. I will revert for the sake of keeping the PR focused on the change. I presume that's what you're trying to say.

Reply:
Reusing code is fine. Just wanted to make sure that the changeset doesn't have unrelated refactoring like the ones above